dot/rpc/modules/childstate_integration_test.go

```diff
@@ -0,0 +1,298 @@
+	invalidBlockHash := common.BytesToHash([]byte("invalid block hash"))
+
+	tests := []struct {
+		expect   string
+		err      error
+		hash     *common.Hash
+		keyChild []byte
+		entry    []byte
+	}{
+		{
+			err:      nil,
+			expect:   common.BytesToHash([]byte(":child_first_value")).String(),
+			hash:     nil,
+			entry:    []byte(":child_first"),
+			keyChild: []byte(":child_storage_key"),
+		},
+		{
+			err:      nil,
+			expect:   common.BytesToHash([]byte(":child_second_value")).String(),
+			hash:     &blockHash,
+			entry:    []byte(":child_second"),
+			keyChild: []byte(":child_storage_key"),
+		},
+		{
+			err:      fmt.Errorf("child trie does not exist at key %s%s", trie.ChildStorageKeyPrefix, []byte(":not_exist")),
+			hash:     &blockHash,
+			entry:    []byte(":child_second"),
+			keyChild: []byte(":not_exist"),
+		},
+		{
+			err:  chaindb.ErrKeyNotFound,
+			hash: &invalidBlockHash,
+		},
+	}
+
+	for _, test := range tests {
+		var req GetStorageHash
+		var res string
+
+		req.Hash = test.hash
+		req.EntryKey = test.entry
+		req.KeyChild = test.keyChild
+
+		err := mod.GetStorageHash(nil, &req, &res)
+
+		if test.err != nil {
+			require.Error(t, err)
+			require.Equal(t, err, test.err)
+		} else {
+			require.NoError(t, err)
+		}
+
+		if test.expect != "" {
+			require.Equal(t, test.expect, res)
+		}
+	}
+}
+
+func TestGetChildStorage(t *testing.T) {
+	mod, blockHash := setupChildStateStorage(t)
+	randomHash, err := common.HexToHash(RandomHash)
+	require.NoError(t, err)
+
+	testCases := []struct {
+		params   []string
+		expected []byte
+		errMsg   string
+	}{
+		{params: []string{":child_storage_key", ""}, expected: nil},
+		{params: []string{":child_storage_key", ":child_first"}, expected: []byte(":child_first_value")},
+		{params: []string{":child_storage_key", ":child_first", blockHash.String()}, expected: []byte(":child_first_value")},
+		{params: []string{":child_storage_key", ":child_first", randomHash.String()}, errMsg: "Key not found"},
+	}
+
+	for _, test := range testCases {
+		t.Run(fmt.Sprintf("%s", test.params), func(t *testing.T) {
+			var res StateStorageResponse
+			var req ChildStateStorageRequest
+
+			if test.params[0] != "" {
+				req.ChildStorageKey = []byte(test.params[0])
+			}
+
+			if test.params[1] != "" {
+				req.Key = []byte(test.params[1])
+			}
+
+			if len(test.params) > 2 && test.params[2] != "" {
+				req.Hash = &common.Hash{}
+				*req.Hash, err = common.HexToHash(test.params[2])
+				require.NoError(t, err)
+			}
+
+			err = mod.GetStorage(nil, &req, &res)
+			// Handle error cases.
+			if test.errMsg != "" {
+				require.Error(t, err)
+				require.Equal(t, err.Error(), test.errMsg)
+				return
+			}
+
+			// Verify expected values.
+			require.NoError(t, err)
+			if test.expected != nil {
+				// Convert human-readable result value to hex.
+				expectedVal := "0x" + hex.EncodeToString(test.expected)
+				require.Equal(t, StateStorageResponse(expectedVal), res)
+			}
```

Comment:
Should we check `StateStorageResponse(expectedVal)` is also as expected when `test.expected` is `nil`?